sdk/servicebus/Azure.Messaging.ServiceBus/src/Sender/ServiceBusSenderClient.cs

```diff
@@ -0,0 +1,522 @@
+        ///
+        /// <returns>A task to be resolved on when the operation has completed.</returns>
+        ///
+        /// <seealso cref="SendRangeInternal(IEnumerable{ServiceBusMessage}, CancellationToken)"/>
+        ///
+        public virtual async Task SendAsync(
+            ServiceBusMessage message,
+            CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(message, nameof(message));
+            await SendRangeAsync(new ServiceBusMessage[] { message }, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        ///   Sends a set of events to the associated Service Bus entity using a batched approach.  If the size of events exceed the
+        ///   maximum size of a single batch, an exception will be triggered and the send will fail.
+        /// </summary>
+        ///
+        /// <param name="messages">The set of event data to send.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        /// <returns>A task to be resolved on when the operation has completed.</returns>
+        ///
+        /// <seealso cref="SendRangeInternal(IEnumerable{ServiceBusMessage}, CancellationToken)"/>
+        ///
+        public virtual async Task SendRangeAsync(IEnumerable<ServiceBusMessage> messages, CancellationToken cancellationToken = default) =>
+            await SendRangeInternal(messages, cancellationToken).ConfigureAwait(false);
+
+        /// <summary>
+        /// Schedules a message to appear on Service Bus at a later time.
+        /// </summary>
+        /// <param name="message"></param>
+        /// <param name="scheduleEnqueueTimeUtc">The UTC time at which the message should be available for processing</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>The sequence number of the message that was scheduled.</returns>
+        public virtual async Task<long> ScheduleMessageAsync(ServiceBusMessage message, DateTimeOffset scheduleEnqueueTimeUtc, CancellationToken cancellationToken = default)
+        {
+            //this.ThrowIfClosed();
+            message.ScheduledEnqueueTimeUtc = scheduleEnqueueTimeUtc.UtcDateTime;
+            return await Connection.ScheduleMessageAsync(message, RetryPolicy, GetSendLinkName(), cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Cancels a message that was scheduled.
+        /// </summary>
+        /// <param name="sequenceNumber">The <see cref="ServiceBusMessage.SystemPropertiesCollection.SequenceNumber"/> of the message to be cancelled.</param>
+        /// <param name="cancellationToken"></param>
+        public virtual async Task CancelScheduledMessageAsync(long sequenceNumber, CancellationToken cancellationToken = default)
+        {
+            //this.ThrowIfClosed();
```

Comment:
Do you think we should add a null check here for `sequenceNumber`?

Reply:
It's a non-nullable value type so null can't be passed.